contracts/Vesting.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./DPT.sol";
+import "hardhat/console.sol";
+
+/// @title Vesting
+/// @author Eldar Dryga
+/// @notice this contract allows you to distribute tokens to users within a certain time
+
+contract Vesting is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    IERC20 private immutable _token;
+
+    uint256 public periodOfDistributionInDays;
+    uint256 public amountOfDistributionPerUser;
+    bool public initialized;
+    uint256 public users;
+    uint256 public maxUsers;
+
+
+    struct UsersInfo {
+        uint256 userTime;
```

Comment:
[Learn this](https://dev.to/javier123454321/solidity-gas-optimizations-pt-3-packing-structs-23f4)
Simply saying you can optimise structure here to use only 2 storage slots

Reply:
Still 4 slots, can be 2 slots